server/models/Message.js

```diff
@@ -275,8 +275,34 @@ const getMessages = async (session, filterCriteria = undefined) => {
   );
 };
 
+const getSurveyReport = async (messageRepository, surveyId) => {
+  const surveyResponses = await messageRepository.getSurveyResponse(surveyId);
+  const surveyReports = surveyResponses.reduce((a, c) => {
+    const { survey_response } = c.survey_response;
+    for (let i = 0; i < survey_response.length; i++) {
+      const e = survey_response[i];
+      if (a[i] === undefined) { a[i] = {}; }
+      if (a[i][e] === undefined) { a[i][e] = 0; }
+      a[i][e] += 1;
+    };
+    return a;
+  }, [])
+    .map(counter => {
+      return {
+        labels: [...Object.keys(counter)],
+        datasets: [{
+          label: "-",
+          data: [...Object.values(counter)],
+        }]
+      }
+    });
+
+  return surveyReports;
+}
+
```

Comment:
This should go into the Survey model, as it does not require any information about the Message object